src/bundle/makeRepairDataProperties.js

```diff
@@ -59,89 +57,74 @@ export default function makeRepairDataProperties() {
+              enumerable: desc.enumerable,
+              configurable: desc.configurable,
+            },
           });
         }
       }
 
-      defineProperty(obj, prop, {
-        get: getter,
-        set: setter,
-        enumerable: desc.enumerable,
-        configurable: desc.configurable,
+      defineProperties(obj, {
+        [prop]: {
+          get: getter,
+          set: setter,
+          enumerable: desc.enumerable,
+          configurable: desc.configurable,
+        },
       });
     }
   }
 
-  /**
-   * These properties are subject to the override mistake
-   * and must be converted before freezing.
-   */
-  function repairDataProperties(intrinsics) {
-    const { global: g, anonIntrinsics: a } = intrinsics;
-
-    const toBeRepaired = [
-      g.Object.prototype,
-      g.Array.prototype,
-      // g.Boolean.prototype,
-      // g.Date.prototype,
-      // g.Number.prototype,
-      // g.String.prototype,
-      // g.RegExp.prototype,
-
-      g.Function.prototype,
-      a.GeneratorFunction.prototype,
-      a.AsyncFunction.prototype,
-      a.AsyncGeneratorFunction.prototype,
-
-      a.IteratorPrototype,
-      // a.ArrayIteratorPrototype,
-
-      // g.DataView.prototype,
-
-      a.TypedArray.prototype,
-      // g.Int8Array.prototype,
-      // g.Int16Array.prototype,
-      // g.Int32Array.prototype,
-      // g.Uint8Array.prototype,
-      // g.Uint16Array.prototype,
-      // g.Uint32Array.prototype,
-
-      g.Error.prototype,
-      // g.EvalError.prototype,
-      // g.RangeError.prototype,
-      // g.ReferenceError.prototype,
-      // g.SyntaxError.prototype,
-      // g.TypeError.prototype,
-      // g.URIError.prototype,
-    ];
+  function repairOneProperty(obj, prop) {
+    const desc = getOwnPropertyDescriptor(obj, prop);
+    if (!desc) {
+      return;
+    }
+    enableDerivedOverride(obj, prop, desc);
+  }
 
-    // Promise may be removed from the whitelist
-    // TODO: the toBeRepaired list should be prepared
-    // externally and provided to repairDataProperties
-    const PromisePrototype = g.Promise && g.Promise.prototype;
-    if (PromisePrototype) {
-      toBeRepaired.push(PromisePrototype);
+  function repairAllProperties(obj) {
+    const descs = getOwnPropertyDescriptors(obj);
+    if (!descs) {
+      return;
     }
+    ownKeys(descs).forEach(prop =>
+      enableDerivedOverride(obj, prop, descs[prop]),
+    );
+  }
 
-    // repair each entry
-    toBeRepaired.forEach(obj => {
-      if (!obj) {
-        return;
-      }
-      const descs = getOwnPropertyDescriptors(obj);
-      if (!descs) {
-        return;
+  function walkRepairPlan(obj, plan) {
+    if (!obj) {
+      return;
+    }
+    ownKeys(plan).forEach(prop => {
+      const subPlan = plan[prop];
+      const subObj = obj[prop];
+      switch (subPlan) {
+        case true:
+          repairOneProperty(obj, prop);
+          break;
+
+        case '*':
+          repairAllProperties(subObj);
```

Comment:
Should only `repairAllProperties` when `subObj` is truthy, as `getOwnPropertyDescriptors` throws `TypeError` if `subObj` is `null` or `undefined`.

Reply:
On line 94 above, at the beginning of `repairAllProperties`, I see

```js
if (!obj) {
  return;
}
```

so I don't think this can be a problem.

Reply:
My comment was addressed by a later change.